servicetalk-buffer-netty/src/main/java/io/servicetalk/buffer/netty/UnreleasableDirectByteBuf.java

```diff
@@ -44,31 +43,11 @@
         super.retain();
     }
 
-    @Override
-    public ByteBuf asReadOnly() {
-        return Unpooled.unreleasableBuffer(super.asReadOnly());
-    }
-
-    @Override
-    public ByteBuf readSlice(int length) {
-        return Unpooled.unreleasableBuffer(super.readSlice(length));
-    }
-
     @Override
     public ByteBuf readRetainedSlice(int length) {
         return readSlice(length);
```

Comment:
Actually, we can also remove these methods that are here to avoid an additional call to `retain()`. As `retain()` it just noon for our implementations, it's ok to use `readRetainedSlice` from `AbstractByteBuf`. Tests are all green if I remove these methods. WDYT?

Reply:
(I am not sure if you are saying the same thing)

I guess it was here before since this was calling `readSlice()` which was wrapping into the `unreleasableBuffer`. Since `readSlice()` override is removed, this becomes redundant.